### PR TITLE
mindustry: init at 99 (and updates)

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by wamserma** [comment]
LGTM. Tested on NixOS 19.09.

**Comment by MicrosoftTakeover** [comment]
LGTM

File: pkgs/games/mindustry/default.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    '';
+    # perl code mavenizes pathes (com.squareup.okio/okio/1.13.0/a9283170b7305c8d92d25aff02a6ab7e45d06cbe/okio-1.13.0.jar -> com/squareup/okio/okio/1.13.0/okio-1.13.0.jar)
+    installPhase = ''
+      find $GRADLE_USER_HOME/caches/modules-2 -type f -regex '.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+    '';
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "1s9rxvc4hx2b9nr9kkay4163rl5fmw80diwj8dkybxxqd1axvdpg";
```
Comment (tadfisher) [inline pkgs/games/mindustry/default.nix]: This hash will change whenever one of the plugins with SNAPSHOT versions adds a new artifact. You may want to pin those by patching the build scripts.

Comment (fgaz) [reply pkgs/games/mindustry/default.nix]: Pinned packr (robovm gets removed).

**Comment by fgaz** [comment]
v100 is out, I just pushed an update. Is there anything blocking this?

File: pkgs/games/mindustry/default.nix [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    type = "Application";
+    name = "Mindustry";
+    desktopName = "Mindustry";
+    exec = "mindustry";
+    icon = "mindustry";
+  };
+
+  postPatch = ''
+    # Remove unbuildable iOS stuff
+    sed -i '163,196d' build.gradle
```
Comment (chkno) [inline pkgs/games/mindustry/default.nix]: Use `sed -i '/^project(":ios"){/,/^}/d' build.gradle` so these line numbers don't need to be updated with every version bump.

Comment (fgaz) [reply pkgs/games/mindustry/default.nix]: Great idea, thank you! Updated.

**Comment by fgaz** [comment]
Updated to 101

**Comment by fgaz** [comment]
Updated to 101.1

...yes, I'll continue to ping everyone here at every release until this gets merged or rejected ;-)

**Comment by jappeace** [comment]
please merge this

**Comment by fgaz** [comment]
Updated to 102 and rebased

**Comment by greizgh** [comment]
Built and tested successfully on x86_64

**Review by ghost — APPROVED** [review]
I think this package has above-average quality and can be merged. I have built it using sandboxing on NixOS x86_64 and tested the basic functionality.

- One small wish/suggestion would be to package the server as a seperate derivation, to prevent pulling in pulseaudio on a headless machine, but I can also propose the change after it was merged.

- Also I noticed it does not seem to like SDL_VIDEODRIVER=wayland, which I have set in my environment, but this is a problem for upstream. It might be good to add set SDL_VIDEODRIVER to x11 explicitly in the wrapper for users who have set it to wayland in their environment (because most applications just work with it).
 
  ```
  [I] Initialized Discord rich presence.
  arc.backend.sdl.SDLError: wayland not available
  	at arc.backend.sdl.SdlApplication.check(SdlApplication.java:181)
  	at arc.backend.sdl.SdlApplication.init(SdlApplication.java:80)
  	at arc.backend.sdl.SdlApplication.<init>(SdlApplication.java:30)
  	at mindustry.desktop.DesktopLauncher.main(DesktopLauncher.java:46)
  [I] Sending crash report.
  [I] Crash sent successfully.
  ```

  This is not critical though because every person who has set this variable manually in their setup will know what to do from the error message.

- You may want to squash all of those commits together.

**Comment by fgaz** [comment]
@petabyteboy 
I reported the wayland issue upstream: https://github.com/Anuken/Mindustry/issues/1393
Splitting it is a good idea. Opened #77893 to track it.

**Pull request merged**